advance-exercises/src/components/03-HigherOrder.js

```diff
@@ -1,15 +1,37 @@
-/*
-  Exercise:
+/* eslint-disable react/no-unused-state */
 
-  Make `withMouse` a "higher-order component" that sends the mouse position
-  to the component as props (hint: use `event.clientX` and `event.clientY`).
-
-*/
 import React from 'react';
 import PropTypes from 'prop-types';
 
 function withMouse(Component) {
-  return Component;
+  class WithMouse extends React.Component {
```

Comment:
Is it good practice to name the inner class with the same name as the enclosing function?